src/cpp/redis.cpp

```diff
@@ -320,13 +293,59 @@ CommandReply Redis::get_model(const std::string& key)
 
 CommandReply Redis::get_script(const std::string& key)
 {
-    Command cmd;
+    SingleKeyCommand cmd;
     cmd.add_field("AI.SCRIPTGET");
     cmd.add_field(key, true);
     cmd.add_field("SOURCE");
     return this->run(cmd);
 }
 
+inline CommandReply Redis::_run(const Command& cmd)
+{
```

Comment:
Hmm. We're getting the polymorphism with the different run overloads and then collapsing it back to a single `_run()` command. Is this actually buying us anything?

Reply:
If not, not need to declare and implement the various overloads. It looks like it matters more for RedisCluster, so you can have the overloads there.

Reply:
the polymorphic functions are inherited from `RedisServer`, but are more useful for `RedisCluster` than `Redis`.  Right now, the `_run()` command is in `Redis` to avoid code repetition since all command types run the same way if they aren't being run on a cluster.